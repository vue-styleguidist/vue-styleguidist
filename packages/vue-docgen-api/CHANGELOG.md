# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [3.11.2](https://github.com/vue-styleguidist/vue-styleguidist/compare/v3.11.1...v3.11.2) (2019-03-28)

**Note:** Version bump only for package vue-docgen-api





## [3.11.1](https://github.com/vue-styleguidist/vue-styleguidist/compare/v3.11.0...v3.11.1) (2019-03-28)


### Bug Fixes

* **docgen:** make displayName tag work ([#358](https://github.com/vue-styleguidist/vue-styleguidist/issues/358)) ([c3e12ce](https://github.com/vue-styleguidist/vue-styleguidist/commit/c3e12ce)), closes [#357](https://github.com/vue-styleguidist/vue-styleguidist/issues/357)





# [3.11.0](https://github.com/vue-styleguidist/vue-styleguidist/compare/v3.10.2...v3.11.0) (2019-03-26)

**Note:** Version bump only for package vue-docgen-api
