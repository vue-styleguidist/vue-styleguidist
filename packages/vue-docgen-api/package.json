{
  "name": "vue-docgen-api",
  "version": "3.11.2",
  "description": "Toolbox to extract information from Vue component files for documentation generation purposes.",
  "main": "dist/main.js",
  "types": "dist/main.d.ts",
  "scripts": {
    "compile": "tsc -p ./tsconfig.build.json"
  },
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/vue-styleguidist/vue-styleguidist.git"
  },
  "bugs": {
    "url": "https://github.com/vue-styleguidist/vue-styleguidist/issues"
  },
  "keywords": [
    "vue",
    "documentation-generation",
    "jsdoc",
    "parse"
  ],
  "author": "Rafael Escala",
  "license": "MIT",
  "dependencies": {
    "@babel/parser": "^7.2.3",
    "@babel/types": "^7.0.0",
    "ast-types": "^0.12.2",
    "hash-sum": "^1.0.2",
    "lru-cache": "^4.1.5",
    "pug": "^2.0.3",
    "recast": "^0.17.3",
    "ts-map": "^1.0.3",
    "typescript": "^3.2.2",
    "vue": "^2.6.9",
    "vue-template-compiler": "^2.6.10"
  },
  "devDependencies": {
    "vue-property-decorator": "^7.3.0"
  }
}
