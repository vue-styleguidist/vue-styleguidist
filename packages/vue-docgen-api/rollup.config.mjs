import commonjs from '@rollup/plugin-commonjs'
import resolve from '@rollup/plugin-node-resolve'
import typescript from '@rollup/plugin-typescript'
import pkg from './package.json' assert { type: 'json' }

const deps = { ...pkg.dependencies }
delete deps['ts-map']

export default {
	input: './src/main.ts',
	output: [
		{
			file: `./dist/main.cjs`,
			format: 'cjs',
      sourcemap: true
		}
	],
	plugins: [
		resolve(),
		commonjs(),
		typescript({
			tsconfig: './tsconfig.build.json',
			declaration: false,
      sourceMap: true,
		})
	],
	external: Object.keys(deps)
}
