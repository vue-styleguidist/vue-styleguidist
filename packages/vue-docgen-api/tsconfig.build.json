{
	"extends": "../../tsconfig.json",
	"compilerOptions": {
		"outDir": "dist",
		"declaration": true,
		"sourceMap": false,
		"skipLibCheck": true
	},
	"include": ["src/**/*.ts", "../../@types/**/*"],
	"exclude": ["src/**/*.spec.ts"]
}
