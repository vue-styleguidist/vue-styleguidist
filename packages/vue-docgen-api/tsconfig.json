{
	"extends": "../../tsconfig.json",
	"compilerOptions": {
		"target": "ES5",
		"outDir": "dist",
		"declaration": true,
		"sourceMap": false,
		"downlevelIteration": true,
		"skipLibCheck": true
	},
	"include": ["src/**/*.ts", "tests/**/*.ts", "../../@types/*.ts"]
}
