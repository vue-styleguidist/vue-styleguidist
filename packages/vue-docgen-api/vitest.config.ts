import { defineConfig } from 'vitest/config'
import globalConfig from '../../vitest.config'

export default defineConfig({
	...globalConfig,
	test: {
		...(globalConfig as any).test,
		setupFiles: ['./tests/setup.ts']
	}
})
