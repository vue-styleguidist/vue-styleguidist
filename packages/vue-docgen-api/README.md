# vue-docgen-api

[Online Documentation](https://vue-styleguidist.github.io/docs/Docgen.html)

`vue-docgen-api` is a toolbox to help extracting information from [Vue](https://vuejs.org/) components, and generate documentation from it.

Use [@babel/parser](https://babeljs.io/docs/en/babel-parser) to parse the code and analyze the contents of the component extracting methods, props events and slots. The output is a JavaScript object.

## Install

Install the module directly from npm:

```sh
npm install vue-docgen-api --save-dev
```

## API

The tool can be used programmatically to extract component information and customize the extraction process:

```js
var vueDocs = require('vue-docgen-api')
var componentInfo
vueDocs.parse(filePath).then(ci => {
  componentInfo = ci
})
```

or with typescript

```ts
import { parse } from 'vue-docgen-api'

async function parseMyComponent(filePath: string) {
  var componentInfoSimple = await parse(filePath)
  var componentInfoConfigured = await parse(filePath, {
    alias: { '@assets': path.resolve(__dirname, 'src/assets') },
    modules: [path.resolve(__dirname, 'src')],
    addScriptHandler: [
      function(
        documentation: Documentation,
        componentDefinition: NodePath,
        astPath: bt.File,
        opt: ParseOptions
      ) {
        // handle custom code in script
      }
    ],
    addTemplateHandler: [
      function(
        documentation: Documentation,
        templateAst: ASTElement,
        options: TemplateParserOptions
      ) {
        // handle custom directives here
      }
    ],
    preScriptHandlers: [
      function(
        documentation: Documentation,
        componentDefinition: NodePath,
        astPath: bt.File,
        opt: ParseOptions
      ) {
        // replaces handlers run before the scriptHandlers
      }
    ],
    scriptHandlers: [
      function(
        documentation: Documentation,
        componentDefinition: NodePath,
        astPath: bt.File,
        opt: ParseOptions
      ) {
        // replaces all the scriptHandlers
      }
    ],
    templateHandlers: [
      function(
        documentation: Documentation,
        templateAst: ASTElement,
        options: TemplateParserOptions
      ) {
        // replaces all the templateHandlers
      }
    ],
    validExtends: (fullFilePath: string) =>
      /[\\/]@my-component-library[\\/]/.test(fullFilePath) ||
      !/[\\/]node_modules[\\/]/.test(fullFilePath)
    jsx: true,
    nameFilter: ['MyComponent']
  })
}
```

### `parse()` vs `parseMulti()`

The API exports two asynchronous functions `parse` and `parseMulti`. The two functions take the same set of parameters. `parse` returns a pure `ComponentDoc` whereas `parseMulti` returns an array of `ComponentDoc`

When using only SFC or components that are the only one in their components, this function returns a `ComponentDoc` object. Using `parse` in most cases is simpler.

If you are creating a library and your goal is to have it as generic as possible, it might be a good idea to use `parseMulti`. In some cases, developer choose to have **more than one component** exported by a file, which make `parse` throw an error.

### Prototypes

```ts
parse(filePath: string, options: DocGenOptions): Promise<ComponentDoc>;
parseSource(source: string, filePath: string, opts?: DocGenOptions): Promise<ComponentDoc>;
parseMulti(filePath: string, options: DocGenOptions): Promise<ComponentDoc[]>;
```

### options `DocGenOptions`

#### `alias`

This is a mirror to the [wepbpack alias](https://webpack.js.org/configuration/resolve/#resolvealias) options. If you are using [alias in Webpack](https://webpack.js.org/configuration/resolve/#resolvealias) or paths in TypeScript, you should reflect this here..

#### `modules`

`modules` mirrors the [webpack option](https://webpack.js.org/configuration/resolve/#resolvemodules) too. If you have it in webpack or use `baseDir` in your tsconfig.json, you should probably see how this one works.

#### `addScriptHandler` and `addTemplateHandler`

The custom additional handlers allow you to add custom handlers to the parser. A handler can navigate and see custom objects that the standard parser would ignore.

#### `preScriptHandlers`, `scriptHandlers` and `templateHandlers`

Replaces all of the handlers by those specified. If each of those 3 `handlers` are set to [], the library will only parse the given component. It will not run any standard handlers anymore.

> **NOTE** Standard handlers are available as namespaces. Import and use them this way:
>
> ```js
> import {
>   parse,
>   ScriptHandlers,
>   TemplateHandlers
> } from 'vue-docgen-api'
>
> parse('myComp', {
>   scriptHandlers: [ScriptHandlers.componentHandler],
>   templateHandlers: [TemplateHandlers.slotHandler]
> })
> ```

#### `validExtend`

Function - Returns if an extended component should be parsed by docgen.

**NOTE** If docgen fails to parse the targetted component, it will log a warning. It is non-blocking but annoying.

**NOTE** If you allow all of `node_modules` to try to be parsed, you might kill performance. Use responsibly.

#### `jsx`

Does your component contain JSX? By default, this is set to false to avoid conflicts with the <> syntax from TypeScript.

If it does, set this flag to true.

#### `nameFilter`

If a file exports multiple components and you only want one, use this option to filter the named exports.

It is noticeably useful when browsing through extended components and mixins. If left blank (undefined), will look at all exports

## Using JSDoc tags

You can use JSDoc tags when documenting components, props and methods.

## Props

```js
export default {
  props: {
    /**
     * Color of the button.
     */
    color: {
      type: String,
      default: '#FCC'
    },
    /**
     * initial value to be passed but undocumented
     * @ignore
     */
    initialvalue: {
      type: Number,
      default: 0
    },
    /**
     * The size of the button allows only some values
     * @values small, medium, large
     */
    size: {
      default: 'normal'
    }
  }
}
```

## Events

```js
export default {
  methods: {
    emitSuccess() {
      /**
       * Success event.
       *
       * @event success
       * @property {string} content content of the first prop passed to the event
       * @property {object} example the demo example
       */
      this.$emit('success', content, {
        demo: 'example'
      })
    }
  }
}
```

## Slots

```html
<template>
  <div>
    <!-- @slot Use this slot header -->
    <slot name="header"></slot>

    <!--
      @slot Modal footer
      @binding item an item passed to the footer
		-->
    <slot name="footer" :item="item" />
  </div>
</template>
```

## Full Example

For the following component

```vue
<template>
  <div>
    <!-- @slot Use this slot header -->
    <slot name="header"></slot>

    <table class="grid">
      <!-- -->
    </table>

    <!--
      @slot Modal footer
      @binding item an item passed to the footer
		-->
    <slot name="footer" :item="item" />
  </div>
</template>

<script>
import { text } from './utils'

/**
 * This is an example of creating a reusable grid component and using it with external data.
 * @version 1.0.5
 * @author [Rafael](https://github.com/rafaesc92)
 * @since Version 1.0.1
 */
export default {
  name: 'grid',
  props: {
    /**
     * object/array defaults should be returned from a factory function
     * @version 1.0.5
     * @since Version 1.0.1
     * @see See [Wikipedia](https://en.wikipedia.org/wiki/Web_colors#HTML_color_names) for a list of color names
     * @link See [Wikipedia](https://en.wikipedia.org/wiki/Web_colors#HTML_color_names) for a list of color names
     */
    msg: {
      type: [String, Number],
      default: text
    },
    /**
     * Model example
     * @model
     */
    value: {
      type: String
    },
    /**
     * describe data
     * @version 1.0.5
     */
    data: [Array],
    /**
     * get columns list
     */
    columns: [Array],
    /**
     * filter key
     * @ignore
     */
    filterKey: {
      type: String,
      default: 'example'
    }
  },
  data() {
    var sortOrders = {}
    this.columns.forEach(function(key) {
      sortOrders[key] = 1
    })
    return {
      sortKey: '',
      sortOrders: sortOrders
    }
  },
  computed: {
    filteredData: function() {
      var sortKey = this.sortKey
      var filterKey = this.filterKey && this.filterKey.toLowerCase()
      var order = this.sortOrders[sortKey] || 1
      var data = this.data
      if (filterKey) {
        data = data.filter(function(row) {
          return Object.keys(row).some(function(key) {
            return (
              String(row[key])
                .toLowerCase()
                .indexOf(filterKey) > -1
            )
          })
        })
      }
      if (sortKey) {
        data = data.slice().sort(function(a, b) {
          a = a[sortKey]
          b = b[sortKey]
          return (a === b ? 0 : a > b ? 1 : -1) * order
        })
      }
      return data
    }
  },
  filters: {
    capitalize: function(str) {
      return str.charAt(0).toUpperCase() + str.slice(1)
    }
  },
  methods: {
    /**
     * Sets the order
     *
     * @public
     * @version 1.0.5
     * @since Version 1.0.1
     * @param {string} key Key to order
     * @returns {string} Test
     */
    sortBy: function(key) {
      this.sortKey = key
      this.sortOrders[key] = this.sortOrders[key] * -1

      /**
       * Success event.
       *
       * @event success
       * @type {object}
       */
      this.$emit('success', {
        demo: 'example'
      })
    },

    hiddenMethod: function() {}
  }
}
</script>
```

we are getting this output:

```js
const componentDoc = {
  displayName: 'grid',
  description:
    'This is an example of creating a reusable grid component and using it with external data.',
  tags: {
    version: [
      {
        description: '1.0.5',
        title: 'version'
      }
    ],
    author: [
      {
        description: '[Rafael](https://github.com/rafaesc92)',
        title: 'author'
      }
    ],
    since: [
      {
        description: 'Version 1.0.1',
        title: 'since'
      }
    ]
  },
  exportName: 'default',
  props: [
    {
      description:
        'object/array defaults should be returned from a factory function',
      tags: {
        version: [
          {
            description: '1.0.5',
            title: 'version'
          }
        ],
        since: [
          {
            description: 'Version 1.0.1',
            title: 'since'
          }
        ],
        see: [
          {
            description:
              'See [Wikipedia](https://en.wikipedia.org/wiki/Web_colors#HTML_color_names) for a list of color names',
            title: 'see'
          }
        ],
        link: [
          {
            description:
              'See [Wikipedia](https://en.wikipedia.org/wiki/Web_colors#HTML_color_names) for a list of color names',
            title: 'link'
          }
        ]
      },
      name: 'msg',
      type: {
        name: 'string|number'
      },
      defaultValue: {
        func: false,
        value: 'text'
      }
    },
    {
      description: 'Model example',
      tags: {
        model: [
          {
            description: true,
            title: 'model'
          }
        ]
      },
      name: 'v-model',
      type: {
        name: 'string'
      }
    },
    {
      description: 'describe data',
      tags: {
        version: [
          {
            description: '1.0.5',
            title: 'version'
          }
        ]
      },
      name: 'data',
      type: {
        name: 'array'
      }
    },
    {
      description: 'get columns list',
      tags: {},
      name: 'columns',
      type: {
        name: 'array'
      }
    },
    {
      description: 'filter key',
      tags: {
        ignore: [
          {
            description: true,
            title: 'ignore'
          }
        ]
      },
      name: 'filterKey',
      type: {
        name: 'string'
      },
      defaultValue: {
        func: false,
        value: "'example'"
      }
    }
  ],
  events: [
    {
      name: 'success',
      description: 'Success event.',
      type: {
        names: ['object']
      }
    }
  ],
  methods: [
    {
      name: 'sortBy',
      modifiers: [],
      description: 'Sets the order',
      tags: {
        access: [
          {
            description: 'public',
            title: 'access'
          }
        ],
        version: [
          {
            description: '1.0.5',
            title: 'version'
          }
        ],
        since: [
          {
            description: 'Version 1.0.1',
            title: 'since'
          }
        ],
        params: [
          {
            title: 'param',
            type: {
              name: 'string'
            },
            name: 'key',
            description: 'Key to order'
          }
        ],
        returns: [
          {
            title: 'returns',
            type: {
              name: 'string'
            },
            description: 'Test'
          }
        ]
      },
      params: [
        {
          name: 'key',
          type: {
            name: 'string'
          },
          description: 'Key to order'
        }
      ],
      returns: {
        title: 'returns',
        type: {
          name: 'string'
        },
        description: 'Test'
      }
    }
  ],
  slots: [
    {
      name: 'header',
      description: 'Use this slot header'
    },
    {
      name: 'footer',
      description: 'Modal footer',
      scoped: true,
      bindings: [
        {
          title: 'binding',
          type: {
            name: 'mixed'
          },
          name: 'item',
          description: 'an item passed to the footer'
        }
      ]
    }
  ]
}
```

## Change log

The change log can be found on the [Changelog Page](./CHANGELOG.md).

## Authors and license

[Rafael Escala](https://github.com/rafaesc92)

[Bart Ledoux](https://github.com/elevatebart)

MIT License.
