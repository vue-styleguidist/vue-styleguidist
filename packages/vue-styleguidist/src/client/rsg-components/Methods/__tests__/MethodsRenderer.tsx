import React from 'react'
import PropTypes from 'prop-types'
// import * as path from 'path'
import { render } from '@testing-library/react'
import { parseSource, MethodDescriptor } from 'vue-docgen-api'
import MethodsRenderer, { columns } from '../MethodsRenderer'

// Test renderers with clean readable snapshot diffs
function ColumnsRenderer({ methods }: { methods: MethodDescriptor[] }) {
	return (
		<ul>
			{methods.map((row, rowIdx) => (
				<li key={rowIdx}>
					{columns(methods, {}).map((col, colIdx) => (
						<div key={colIdx}>{col.render(row)}</div>
					))}
				</li>
			))}
		</ul>
	)
}

ColumnsRenderer.propTypes = {
	methods: PropTypes.array
}

async function renderMethodsSection(methods: string[]) {
	const parsed = await parseSource(
		`export default {
			methods: {
				${methods.join(',\n')}
			}
		}`,
		''
	)
	if (Array.isArray(parsed) || !parsed.methods) {
		return render(<div />)
	}
	return render(<ColumnsRenderer methods={parsed.methods} />)
}

describe('MethodsRenderer', () => {
	it('should render a table', () => {
		const actual = render(
			<MethodsRenderer
				methods={[
					{
						name: 'method',
						modifiers: [],
						params: [],
						description: 'Public'
					}
				]}
			/>
		)

		expect(actual.container).toMatchSnapshot()
	})

	it('should render public method', async () => {
		const actual = await renderMethodsSection(['/**\n * Public\n * @public\n */\nmethod() {}'])

		expect(actual.container).toMatchSnapshot()
	})

	it('should render parameters', async () => {
		const actual = await renderMethodsSection([
			'/**\n * Public\n * @public\n * @param {Number} value - Description\n */\nmethod(value) {}'
		])

		expect(actual.container).toMatchSnapshot()
	})

	it('should render returns', async () => {
		const actual = await renderMethodsSection([
			'/**\n * @public\n * @returns {Number} - Description\n */\nmethod() {}'
		])

		expect(actual.container).toMatchSnapshot()
	})

	it('should render JsDoc tags', () => {
		const actual = render(
			<ColumnsRenderer
				methods={[
					{
						name: 'Foo',
						tags: {
							since: [
								{
									title: 'since',
									description: '1.0.0'
								}
							]
						}
					}
				]}
			/>
		)

		expect(actual.container).toMatchSnapshot()
	})

	it('should render deprecated JsDoc tags', () => {
		const actual = render(
			<ColumnsRenderer
				methods={[
					{
						name: 'Foo',
						tags: {
							deprecated: [
								{
									title: 'description',
									description: 'Use *another* method'
								}
							]
						}
					}
				]}
			/>
		)

		expect(actual.container).toMatchSnapshot()
	})
})
