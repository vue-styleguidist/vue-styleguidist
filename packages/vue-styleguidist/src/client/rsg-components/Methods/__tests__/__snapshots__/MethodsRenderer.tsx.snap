// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MethodsRenderer should render JsDoc tags 1`] = `
<div>
  <ul>
    <li>
      <div>
        <code
          class="rsg--name-13"
        >
          Foo()
        </code>
      </div>
      <div>
        <div
          class="vsg-jsdoc-tag rsg--wrapper-16"
        >
          <span
            class="vsg-tag-name rsg--name-17"
          >
            Since
          </span>
          <span
            class="vsg-tag-value undefined"
          >
            <span
              class="rsg--text-22 rsg--inheritSize-23 rsg--baseColor-27"
            >
              1.0.0
            </span>
          </span>
        </div>
      </div>
      <div />
    </li>
  </ul>
</div>
`;

exports[`MethodsRenderer should render a table 1`] = `
<div>
  <table
    class="rsg--table-8"
  >
    <thead
      class="rsg--tableHead-9"
    >
      <tr>
        <th
          class="rsg--cellHeading-10"
        >
          Method name
        </th>
        <th
          class="rsg--cellHeading-10"
        >
          Description
        </th>
        <th
          class="rsg--cellHeading-10"
        >
          Parameters
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="rsg--dataRow-11"
      >
        <td
          class="rsg--cell-12 rsg--name-0"
        >
          <code
            class="rsg--name-13"
          >
            method()
          </code>
        </td>
        <td
          class="rsg--cell-12 rsg--description-3"
        >
          <p
            class="rsg--para-15"
          >
            Public
          </p>
        </td>
        <td
          class="rsg--cell-12"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`MethodsRenderer should render deprecated JsDoc tags 1`] = `
<div>
  <ul>
    <li>
      <div>
        <code
          class="rsg--name-13 rsg--isDeprecated-14"
        >
          Foo()
        </code>
      </div>
      <div>
        <div
          class="vsg-jsdoc-tag rsg--wrapper-16"
        >
          <span
            class="vsg-tag-name rsg--name-17"
          >
            Deprecated
          </span>
          <span
            class="vsg-tag-value undefined"
          >
            <span
              class="rsg--text-22 rsg--inheritSize-23 rsg--baseColor-27"
            >
              Use 
              <em
                class="rsg--text-22 rsg--inheritSize-23 rsg--baseColor-27 rsg--em-29"
              >
                another
              </em>
               method
            </span>
          </span>
        </div>
      </div>
      <div />
    </li>
  </ul>
</div>
`;

exports[`MethodsRenderer should render parameters 1`] = `
<div>
  <ul>
    <li>
      <div>
        <code
          class="rsg--name-13"
        >
          method()
        </code>
      </div>
      <div>
        <p
          class="rsg--para-15"
        >
          Public
        </p>
      </div>
      <div>
        <div
          class="rsg--root-18"
        >
          <div
            class="rsg--block-20"
          >
            <span>
              <code
                class="rsg--name-13"
              >
                value
              </code>
              :
            </span>
             
            <span
              class="rsg--type-21"
            >
              Number
            </span>
             
            —
             
            <span
              class="rsg--text-22 rsg--inheritSize-23 rsg--baseColor-27"
            >
              Description
            </span>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`MethodsRenderer should render public method 1`] = `
<div>
  <ul>
    <li>
      <div>
        <code
          class="rsg--name-13"
        >
          method()
        </code>
      </div>
      <div>
        <p
          class="rsg--para-15"
        >
          Public
        </p>
      </div>
      <div />
    </li>
  </ul>
</div>
`;

exports[`MethodsRenderer should render returns 1`] = `
<div>
  <ul>
    <li>
      <div>
        <code
          class="rsg--name-13"
        >
          method()
        </code>
      </div>
      <div>
        <div>
          Returns:
           
          <span
            class="rsg--type-21"
          >
            Number
          </span>
           
          —
           
          <span
            class="rsg--text-22 rsg--inheritSize-23 rsg--baseColor-27"
          >
            Description
          </span>
        </div>
      </div>
      <div />
    </li>
  </ul>
</div>
`;
